includes/bootstrap.php

```diff
@@ -0,0 +1,23 @@
+<?php
+/**
+ * Central bootstrapping entry point for all non-autoloaded files.
+ *
+ * This file is mainly used for taking direct control of included files off from Composer's
+ * "files" directive, as that one can easily include the files multiple times, leading to
+ * redeclaration fatal errors.
+ *
+ * @package AmpProject/AmpWP
+ */
+
+$files_to_include = [
+	__DIR__ . '/../back-compat/back-compat.php'       => 'amp_backcompat_use_v03_templates',
+	__DIR__ . '/../includes/amp-helper-functions.php' => 'amp_activate',
+	__DIR__ . '/../includes/admin/functions.php'      => 'amp_init_customizer',
+	__DIR__ . '/../includes/deprecated.php'           => 'amp_load_classes',
+];
+
+foreach ( $files_to_include as $file_to_include => $function_to_check ) {
+	if ( ! function_exists( $function_to_check ) ) {
+		include $file_to_include;
+	}
+}
```

Comment:
Wouldn't this be simpler as:

```suggestion
$files_to_include = [
	__DIR__ . '/../back-compat/back-compat.php',
	__DIR__ . '/../includes/amp-helper-functions.php',
	__DIR__ . '/../includes/admin/functions.php',
	__DIR__ . '/../includes/deprecated.php',
];

foreach ( $files_to_include as $file_to_include ) {
	require_once $file_to_include;
}
```

Reply:
Or even simpler:

```suggestion
require_once __DIR__ . '/../back-compat/back-compat.php';
require_once __DIR__ . '/../includes/amp-helper-functions.php';
require_once __DIR__ . '/../includes/admin/functions.php';
require_once  __DIR__ . '/../includes/deprecated.php';
```

Reply:
That could potentially fail in several different scenarios, due to the way the Composer `'files'` directive works.

The only safe behavior is to write the code in such a way that is independent of both loading order and filesystem location. For example, it cannot be assumed that there's only one version of these files, and the `require_once` only checks for the file path, not what it is actually supposed to declare.

Reply:
Got it.